pkg/storage/sasNode.go

```diff
@@ -53,6 +53,75 @@ func (sas *sasStorage) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnsta
 	return nil, status.Error(codes.Unimplemented, "NodeUnstageVolume is not implemented")
 }
 
+// Read the sas address configuration file and return addresses
+func readSASAddrFile() ([]string, error) {
+
+	SASFilePath := "/etc/kubernetes/sas-addresses"
+	file, err := os.Open(SASFilePath)
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	foundAddresses := []string{}
+
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		line = strings.TrimSpace(line)
+		foundAddresses = append(foundAddresses, line)
+	}
+
+	if err := scanner.Err(); err != nil {
+		return nil, err
+	}
+	return foundAddresses, nil
+}
+
+func GetSASInitiators() ([]string, error) {
+	// look for file specifying addresses. Skip discovery process if it exists
+	specifiedSASAddrs, err := readSASAddrFile()
+	if err != nil {
+		klog.Warningf("Error reading sas address config file: %v", err)
+	}
+	if specifiedSASAddrs != nil {
+		return specifiedSASAddrs, nil
+	}
+	klog.Infof("begin SAS address discovery")
+	sasAddrFilename := "host_sas_address"
+	scsiHostBasePath := "/sys/class/scsi_host/"
+
+	dirList, err := os.ReadDir(scsiHostBasePath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, hostDir := range dirList {
+		sasAddrFile := filepath.Join(scsiHostBasePath, hostDir.Name(), sasAddrFilename)
+		addrBytes, err := os.ReadFile(sasAddrFile)
+		address := string(addrBytes)
+		address = strings.TrimLeft(strings.TrimRight(address, "\n"), "0x")
+
+		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				continue
+			} else {
+				klog.Warningf("error searching for SAS HBA addresses: %v", err)
```

Comment:
If this is an error, suggest using, relying on the passed in filename value:
klog.Error(err, "searching for SAS HBA addresses", "filename", filename)